src/npc_class.cpp

```diff
@@ -0,0 +1,122 @@
+#include "npc_class.h"
+#include "debug.h"
+#include "rng.h"
+#include "generic_factory.h"
+
+#include <list>
+
+static const std::array<npc_class_id, 17> legacy_ids = {{
+        npc_class_id( "NC_NONE" ),
+        npc_class_id( "NC_EVAC_SHOPKEEP" ),  // Found in the Evacuation Center, unique, has more goods than he should be able to carry
+        npc_class_id( "NC_SHOPKEEP" ),       // Found in towns.  Stays in his shop mostly.
+        npc_class_id( "NC_HACKER" ),         // Weak in combat but has hacking skills and equipment
+        npc_class_id( "NC_DOCTOR" ),         // Found in towns, or roaming.  Stays in the clinic.
+        npc_class_id( "NC_TRADER" ),         // Roaming trader, journeying between towns.
+        npc_class_id( "NC_NINJA" ),          // Specializes in unarmed combat, carries few items
+        npc_class_id( "NC_COWBOY" ),         // Gunslinger and survivalist
+        npc_class_id( "NC_SCIENTIST" ),      // Uses intelligence-based skills and high-tech items
+        npc_class_id( "NC_BOUNTY_HUNTER" ),  // Resourceful and well-armored
+        npc_class_id( "NC_THUG" ),           // Moderate melee skills and poor equipment
+        npc_class_id( "NC_SCAVENGER" ),      // Good with pistols light weapons
+        npc_class_id( "NC_ARSONIST" ),       // Evacuation Center, restocks moltovs and anarcist type stuff
+        npc_class_id( "NC_HUNTER" ),         // Survivor type good with bow or rifle
+        npc_class_id( "NC_SOLDIER" ),        // Well equiped and trained combatant, good with rifles and melee
+        npc_class_id( "NC_BARTENDER" ),      // Stocks alcohol
+        npc_class_id( "NC_JUNK_SHOPKEEP" )   // Stocks wide range of items...
+    }
+};
+
+npc_class_id NC_NONE( "NC_NONE" );
```

Comment:
Yes, I mostly left them only for compatibility. A lot of code that should be in JSONs is instead in `if( class_id == NC_WHATEVER )` blocks.